Trading/Mode/dip_analyser_trading_mode.py

```diff
@@ -0,0 +1,397 @@
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+
+from ccxt import InsufficientFunds
+
+from config import START_PENDING_EVAL_NOTE, TraderOrderType, TradeOrderSide
+from trading.trader.modes.abstract_mode_creator import AbstractTradingModeCreator
+from trading.trader.modes.abstract_mode_decider import AbstractTradingModeDecider
+from trading.trader.modes.abstract_trading_mode import AbstractTradingMode
+from evaluator.Strategies import DipAnalyserStrategyEvaluator
+from tools.symbol_util import split_symbol
+
+
+class DipAnalyserTradingMode(AbstractTradingMode):
+    DESCRIPTION = "DipAnalyserTradingMode is a trading mode adapted to volatile markets. It will look for local " \
+                  "market bottoms, weight them and buy these bottoms. It never sells except after a buy order is " \
+                  "filled. When a buy order is filled, sell orders will automatically be created at a higher price " \
+                  "than this of the filled buy order. The number of sell orders created after each buy is specified " \
+                  "in DipAnalyserTradingMode.json. A higher risk will make larger buy orders. Make sure to have the " \
+                  "base of each traded pair to be traded otherwise the DipAnalyserTradingMode won't trade: it will " \
+                  "never be able to place the initial buy orders."
+
+    def __init__(self, config, exchange):
+        super().__init__(config, exchange)
+        self.load_config()
+
+    def create_deciders(self, symbol, symbol_evaluator):
+        self.add_decider(symbol, DipAnalyserTradingModeDecider(self, symbol_evaluator, self.exchange))
+
+    def create_creators(self, symbol, symbol_evaluator):
+        self.add_creator(symbol, DipAnalyserTradingModeCreator(self))
+
+    async def order_filled_callback(self, order):
+        decider = self.get_only_decider_key(order.get_order_symbol())
+        await decider.order_filled_callback(order)
+
+
+class DipAnalyserTradingModeCreator(AbstractTradingModeCreator):
+
+    LIMIT_PRICE_MULTIPLIER = 0.995
+    SOFT_MAX_CURRENCY_RATIO = 0.33
+    DEFAULT_FULL_VOLUME = 0.5
+
+    RISK_VOLUME_MULTIPLIER = 0.2
+
+    DELTA_RATIO = 0.8
+
+    VOLUME_WEIGH_TO_VOLUME_PERCENT = {
+       1: 0.5,
+       2: 0.7,
+       3: 1
+    }
+    PRICE_WEIGH_TO_PRICE_PERCENT = {
+       1: 1.04,
+       2: 1.07,
+       3: 1.1
+    }
```

Comment:
Can it be usefull to put this params in a `DipAnalyserTradingModeCreator.json` ?

Reply:
I was thinking about it too when writing it, I don't really know, they are quite complicated settings since they depend on a lot of factors to be chosen

Reply:
Yes, if you think it shouldn't be modified, or it's too hard to modify it's better to keep settings here.

Reply:
Actually i moved them into a config file, this way its easy to adapt to bull/bear markets